fetch-in-react/my-app/src/UserDetails.tsx

```diff
@@ -0,0 +1,43 @@
+/* eslint-disable @typescript-eslint/no-unused-vars -- Remove me */
+import { useEffect, useState } from 'react';
+import { UserCard } from './UserCard';
+import type { User } from './Users';
+
+type Props = {
+  userId: number;
+  onCancel: () => void;
+};
+export function UserDetails({ userId, onCancel }: Props) {
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<unknown>();
+  const [user, setUser] = useState<User>();
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const response = await fetch(
+        `https://jsonplaceholder.typicode.com/users/` + userId
+      );
+      const jsonData = await response.json();
+      setUser(jsonData);
+    };
+    fetchData();
+    setIsLoading(false);
```

Comment:
This should be done inside the `fetchData()` function so that it doesn't get set until after the fetch finishes.